src/beta_inc.jl

```diff
@@ -0,0 +1,945 @@
+     bn = bnp1
+     bnp1 = t
+
+     r0 = r
+     r = anp1/bnp1
+     if abs(r - r0) <= epps*r
+        break
+     end
+     #rescale
+     an /= bnp1
+     bn /= bnp1
+     anp1 = r
+     bnp1 = 1.0
+    end
+    return ans*r
+end
+
+"""
+    basym(a,b,lambda,epps)
+
+Compute ``I_{x}(a,b)`` using asymptotic expansion for a,b >= 15.
+It is assumed that ``\\lambda = (a+b)*y - b``
+"""
+function basym(a::Float64, b::Float64, lambda::Float64, epps::Float64)
+    a0 = b0 = c = d = zeros(21)
+    e0 = 2/sqrt(pi)
+    e1 = 2^(-1.5)
+
+    ans = 0.0
+    if a >= b
+        @goto l10
+    end
+    h = a/b
+    r0 = 1.0/(1.0 + h)
+    r1 = (b-a)/b
+    w0 = 1.0/sqrt(a*(1.0+h))
+    @goto l20
+
+    @label l10
+     h = b/a
+     r0 = 1.0/(1.0 + h)
+     r1 = (b-a)/a
+     w0 = 1.0/sqrt(b*(1.0+h))
+    @label l20
+     f = -a*SpecialFunctions.logmxp1(-lambda/a) + b*SpecialFunctions.logmxp1(lambda/b)
+     t = exp(-f)
+     if t == 0.0
+        return ans
+     end
+     z0 = sqrt(f)
+     z = 0.5*(z0/e1)
+     z² = 2*f
+
+     a0[1] = (2.0/3.0)*r1
+     c[1] = -0.5*a0[1]
+     d[1] = - c[1]
+     j0 = (0.5/e0)*erfcx(z0)
+     j1 = e1
+     sm = j0 + d[1]*w0*j1
+
+     s = 1.0
+     h² = h*h
+     hn = 1.0
+     w = w0
+     znm1 = z
+     zn = z²
+
+     for n = 2:2:20
+        hn *= h²
+        a0[n] = 2.0*r0*(1.0 + h*hn)/(n + 2.0)
+        np1 = n + 1
+        s += hn
+        a0[np1] = 2.0*r1*s/(n+3.0)
+
+        for i = n:np1
+            r = -0.5*(i + 1.0)
+            b0[1] = r*a0[1]
+            for m = 2:i
+                bsum = 0.0
+                mm1 = m - 1
+                for j =1:mm1
+                    mmj = m - j
+                    bsum += (j*r - mmj)*a0[j]*b0[mmj]
+                end
+                b0[m] = r*a0[m] + bsum/m
+            end
+            c[i] = b0[i]/(i + 1.0)
+
+            dsum = 0.0
+            im1 = i - 1
+            for j = 1:im1
+                imj = i - j
+                dsum += d[imj]*c[j]
+            end
+            d[i] = -(dsum + c[i])
+        end
+
+        j0 = e1*znm1 + (n - 1.0)*j0
+        j1 = e1*zn + n*j1
+        znm1 *= z²
+        zn *= z²
+        w *= w0
+        t0 = d[n]*w*j0
+        w *= w0
+        t1 = d[np1]*w*j1
+        sm += t0 + t1
+        if (abs(t0) + abs(t1)) <= epps*sm
+            break
+        end
+     end
+     @label l60
+      u = exp(-bcorr(a,b))
+      return e0*t*u*sm
+end        
+
+"""
+    bgrat(a,b,x,y,w,epps)
+
+Evaluation of I_{x}(a,b) when b < min(epps,epps*a) and x <= 0.5 using asymptotic expansion.
+It is assumed a >= 15 and b <= 1, and epps is tolerance used.
+"""
+function bgrat(a::Float64, b::Float64, x::Float64, y::Float64, w::Float64, epps::Float64)
+    c = zeros(30)
+    d = zeros(30)
+    bm1 = b - 1.0
+    nu = a + 0.5*bm1
+    if y > 0.375
+        @goto l10
+    else
+        lnx = log1p(-y)
+        @goto l11
+    end
+    @label l10
+     lnx = log(x)
+    @label l11
+     z = -nu*lnx
+     if b*z == 0.0
+        @goto l100
```

Comment:
just call `error("....")` for now (give a useful message for why it is happening).